cmd/oras/internal/display/handler.go

```diff
@@ -170,8 +170,11 @@ func NewTagHandler(printer *output.Printer, target option.Target) metadata.TagHa
 }
 
 // NewManifestPushHandler returns a manifest push handler.
-func NewManifestPushHandler(printer *output.Printer) metadata.ManifestPushHandler {
-	return text.NewManifestPushHandler(printer)
+func NewManifestPushHandler(printer *output.Printer, outputDescriptor bool, pretty bool) (status.ManifestPushHandler, metadata.ManifestPushHandler) {
```

Comment:
Should also take descriptor and option.Target as input. Pass descriptor to status handler and pass target to metadata handler